addon/reducer.js

```diff
@@ -1,34 +1,40 @@
 import _ from 'lodash'
 import Ember from 'ember'
 const {Logger} = Ember
-import {CHANGE_VALUE, VALIDATION_RESOLVED} from './actions'
+import {CHANGE_VALUE, VALIDATION_RESOLVED, CHANGE_MODEL} from './actions'
+import convertSchema from './convert-schema'
 
-function ensureParent (state, id) {
+function ensureParent (stateValue, id) {
   // If id does not have a parent the nothing to do
   if (_.isEmpty(id) || id.indexOf('.') === -1) {
     return
   }
 
   const segments = id.split('.')
   const idLastSegment = segments.pop()
-  const relativePath = `value.${segments.join('.')}`
+  const relativePath = segments.join('.')
 
-  const relativeObject = _.get(state, relativePath)
+  const relativeObject = _.get(stateValue, relativePath)
   const isArrayItem = /^\d+$/.test(idLastSegment)
 
   if (isArrayItem && !_.isArray(relativeObject)) {
-    ensureParent(state, segments.join('.'))
-    _.set(state, relativePath, [])
+    ensureParent(stateValue, segments.join('.'))
+    _.set(stateValue, relativePath, [])
   } else if (!isArrayItem && !_.isPlainObject(relativeObject)) {
-    ensureParent(state, segments.join('.'))
-    _.set(state, relativePath, {})
+    ensureParent(stateValue, segments.join('.'))
+    _.set(stateValue, relativePath, {})
   }
 }
 
 const INITIAL_VALUE = {
   errors: {},
   validationResult: {warnings: [], errors: []},
-  value: null
+  value: null,
+  model: {},
+  baseModel: {}
```

Comment:
What is the difference between `model` and `baseModel`? Please add comments here to clarify this.